contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -885,6 +912,164 @@ run_tsql_best_match_heuristics(int nargs, Oid *input_typeids, FuncCandidateList
 	return new_candidates;
 }
 
+/*
+ * For a given function details, validate whether it is in exception function list
+ * and also validate the input argument data types.
+ */
+bool validate_exception_function(char *func_nsname, char *func_name, int nargs, Oid *input_typeids)
+{
+	tsql_exception_function_t  *exception_func = NULL;
+	int							i;
+	bool						type_match;
+	Oid							inputTypId, validTypId;
+	Oid							sys_varcharoid;
+
+	/* Sanity checks */
+	if (func_name == NULL || (nargs != 0 && input_typeids == NULL))
+		return false;
+
+	/* 
+	 * Exception function handling is only for system functions.
+	 * If func_nsname is NULL, consider it to be a "sys".
+	 */
+	if (func_nsname != NULL &&
+		(strlen(func_nsname) != 3 || strncmp(func_nsname, "sys", 3) != 0))
```

Comment:
strlen is required here as `strncmp(func_nsname, "sys", 3)` will compare only first 3 letters of string `func_nsname`. Hence for `func_nsname = "sysabc"`  `strncmp(func_nsname, "sys", 3)` will give zero.

Reply:
If we anyway want to compare the whole strings then why not use ```strcmp```, as it would be equally faster and moreover reduce extra conditional check and overhead on the ```strncmp```.